Forta-Bots-Challenge-3/src/agent.ts

```diff
@@ -0,0 +1,57 @@
+import { BlockEvent, Finding, HandleBlock, getAlerts, getEthersProvider } from "forta-agent";
+import {
+  ETH_CHAIN_ID,
+  L1_ARBITRUM_ESCROW_ADDRESS,
+  L1_DAI_CONTRACT_ADDRESS,
+  L1_OPTIMISM_ESCROW_ADDRESS,
+  L2_DAI_ADDRESS,
+} from "./constants";
+import { providers } from "ethers";
+import { emitEscrowBalance, checkInvariant, contractAddresses, escrowBalances } from "./utils";
+
+let prevEscrowBalances: escrowBalances;
+
+export const provideHandleBlock = (
+  provider: providers.Provider,
+  contractAddresses: contractAddresses,
+  getAlerts: any
+): HandleBlock => {
+  return async (BlockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+    const { chainId } = await provider.getNetwork();
```

Comment:
Please note that for values that can be initialized only once, there's a special `Initialize` block that should be used instead (which should be exported just like `handleBlock`)